pkg/client/cache/reflector.go

```diff
@@ -277,6 +278,7 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			if urlError, ok := err.(*url.Error); ok {
 				if opError, ok := urlError.Err.(*net.OpError); ok {
 					if errno, ok := opError.Err.(syscall.Errno); ok && errno == syscall.ECONNREFUSED {
```

Comment:
I don't disagree, but I'd like to keep this PR focused on just adding log messages so that we can have some visibility into a flake, where it appears that somewhere in the stack we're losing update notifications.